lib/rubycritic/rake_task.rb

```diff
@@ -0,0 +1,57 @@
+require "rake"
+require "rake/tasklib"
+
+module Rubycritic
+  class RakeTask < Rake::TaskLib
+    attr_accessor :name
+    attr_accessor :report
+    attr_accessor :format
+    attr_accessor :paths
+    attr_accessor :options
+
+    def initialize(*args, &task_block)
+      setup_ivars(args)
+
+      desc "Run rubycritic" unless ::Rake.application.last_comment
+
+      task(name, *args) do |_, task_args|
+        RakeFileUtils.send(:verbose, verbose) do
+          if task_block
+            task_block.call(*[self, task_args].slice(0, task_block.arity))
+          end
```

Comment:
This lets you call

```ruby
Rubycritic::RakeTask.new(:mytask, [:version] => :test) do |task|
  task.verbose = true
end
```

See https://github.com/ruby/rake/blob/master/lib/rake/dsl_definition.rb#L28-L67